Makefile

```diff
@@ -17,7 +17,16 @@
 
 CABAL_BIN=cabal
 PWD := $(shell /bin/pwd)
-CABAL = $(CABAL_BIN) --jobs -vnormal+nowrap --project-file=$(PWD)/cabal.project
+
+# There's a lot of parallelism in the schema-generated code
+# If you have >=16G and >=4 cores, trying passing these:
+#
+# EXTRA_GHC_OPTS = '-j4 +RTS -A128m -n2m -RTS'
+#
+EXTRA_GHC_OPTS ?=
+
+CABAL = $(CABAL_BIN) --jobs --ghc-options='$(EXTRA_GHC_OPTS)' \
+            -vnormal+nowrap --project-file=$(PWD)/cabal.project
 
```

Comment:
Passing GHC flags to the cabal-install cli is almost never what you want, since they will almost certainly apply to all the package dependencies, which leads to unnecessary rebuilds.

I would put the `-j` options in the Cabal descriptors where you want to enable parallelism, or in a `cabal.project.local` file like this:
```
package haskell-language-server
  flags: +rename
  ghc-options:
    -eventlog
    -fno-ignore-asserts

package ghcide
  ghc-options:
    -eventlog
    -j4
    -fno-ignore-asserts
```